Modules/IO/NIFTI/test/itkNiftiImageIOTest3.cxx

```diff
@@ -143,6 +143,11 @@ TestImageOfVectors(const std::string & fname)
       }
     }
   }
+  if (!intentCode.empty())
+  {
+    itk::MetaDataDictionary & dictionary = vi->GetMetaDataDictionary();
+    itk::EncapsulateMetaData<std::string>(dictionary, "intent_code", intentCode);
+  }
```

Comment:
The test sets the intent_code before writing, so shouldn't it then also check that after reading the file back, the intent_code is there in the read image?

Reply:
We could defintely add a lot more testing, but that would take much more time. I would prefer to not put much more effort into this change at this point.

Reply:
So then you just test that is it _possible_ to add an entry to the dictionary?

Reply:
The test performs basic checks (there is no crash or exception and the data that is written and read back is the same as the original) when reading and writing displacement fields in NIFTI file.
Probably the next test that someone should add is to compare with some reference NIFTI files. But some time ago I had to add test data and the process was so complicated that I would rather not invest time into going through that again at this point.

Reply:
I was expecting something like this, after `readback = itk::IOTestHelper::ReadImage<VectorImageType>(fname)`:
```
  if (!intentCode.empty())
  {
    const itk::MetaDataDictionary & dictionary = readback->GetMetaDataDictionary();

    std::string readIntentCode;
    if (itk::ExposeMetaData<std::string>(dictionary, "intent_code", readIntentCode)
    {
      if (readIntentCode != intentCode)
      {
        std::cout << "intent_code is different: " << readIntentCode << " != " << intentCode << std::endl;
        same = false;
      }

    }
    else
    {
      std::cout << "The read image should have an intent_code in its dictionary" << std::endl;
      same = false;
    }
  }
```

Isn't that the idea?

Reply:
The main idea is to force to use the displacement field writing/reading branch of the code, which can be triggered by setting this intent code. But adding the suggested test is easy enough, so I've added it now.